tests/test_requests.py

```diff
@@ -858,6 +858,16 @@ def test_prepare_request_with_bytestring_url(self):
         prep = s.prepare_request(req)
         assert prep.url == "https://httpbin.org/"
 
+    def test_request_with_bytestring_host(self):
+        s = requests.Session()
+        resp = s.request(
+            'GET',
+            'http://httpbin.org/cookies/set?cookie=value',
```

Comment:
I know this is already merged, but we may want to amend this test to use the httpbin pytest fixture instead of calling httpbin.org directly. This test will fail if the user doesn't have an internet connection.